### PR TITLE
fix doc for kpt fn doc command

### PR DESCRIPTION
**Pull request by mengqiy**
Also add another test for it.

There are some new comments in https://github.com/GoogleContainerTools/kpt/pull/1556 after it merged. This PR address them.

### CODE REVIEW

File: site/reference/fn/doc/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Doc"
+linkTitle: "doc"
+type: docs
+description: >
+    Get the help text from a function image
```
Comment (frankfarzan) [inline site/reference/fn/doc/README.md:6]: update

File: site/reference/fn/doc/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+<!--mdtogo:Examples-->
+
+```shell
+# diplay the documentation for image gcr.io/kpt-fn/set-namespace:v0.1.1
+kpt fn doc --image gcr.io/kpt-fn/starlark:unstable
+```
+
+<!--mdtogo-->
+
+### Synopsis
```
Comment (frankfarzan) [inline site/reference/fn/doc/README.md:24]: It's fine to say how the function works under the covers in the detail section here.

File: internal/cmdfndoc/cmdfndoc_test.go [hunk 3]
```diff
@@ -30,12 +30,16 @@ func TestFnDoc(t *testing.T) {
 	}
 	testcases := []testcase{
 		{
-			image: "gcr.io/kpt-fn/set-namespace:unstable",
+			image: "gcr.io/kpt-fn/set-namespace:v0.1.1",
 		},
 		{
 			image:     "gcr.io/kpt-fn/set-namespace:v0.1.0",
```
Comment (frankfarzan) [inline internal/cmdfndoc/cmdfndoc_test.go:36]: i'm guessing it's fine if different test cases use different versions of the same function?

File: site/reference/fn/doc/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+kpt fn doc --image gcr.io/kpt-fn/set-namespace:v0.1.1
+```
+
+<!--mdtogo-->
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+`kpt fn doc` invokes `docker run` to extract the help text. It requires the
```
Comment (frankfarzan) [inline site/reference/fn/doc/README.md:28]: docker is a registered trademark and an implementation of container orchestration. Whether kpt invokes docker or not is an implementation detail. What you want to convey ehre:

X invokes the function container with `--help` flag ...

**Review by frankfarzan — CHANGES_REQUESTED** [review]

File: internal/cmdfndoc/cmdfndoc_test.go (hunk 3 above)
Comment (mengqiy) [reply internal/cmdfndoc/cmdfndoc_test.go:36]: gcr.io/kpt-fn/set-namespace:v0.1.0 happens to not work with `kpt fn doc`. We fix that in `v0.1.1`.
I guess this is fine here.

File: site/reference/fn/doc/README.md (hunk 4 above)
Comment (mengqiy) [reply site/reference/fn/doc/README.md]: Updated.

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**